src/NosCore.GameObject/Providers/QuestProvider/QuestProvider.cs

```diff
@@ -248,14 +227,70 @@ public async Task<bool> AddQuestAsync(ICharacterEntity character, short questId)
             }
         }
 
-        public Task ValidateQuest(ClientSession clientSession, Guid characterQuestId)
+        public bool ValidateQuest(ICharacterEntity character, short questId)
         {
-            return Task.CompletedTask;
-            //get quest
-            //check all valid
-            //ApplyBonus
-            //delete objectives
-            //delete quest
+            bool isValid = false;
+            var characterQuest = character.Quests.Values.First(s => s.QuestId == questId);
+            switch (characterQuest.Quest.QuestType)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/manual/NosCoreIO/NosCore/pullRequest?prid=5304333)